duplicate PrivacyInfo.xcprivacy when build YandexMobileAds for iOS

Tested in Defold 1.9.3 and 1.9.4 beta and `build-stage.defold.com` server:

```
com.defold.extender.ExtenderException: java.nio.file.FileAlreadyExistsException: /var/folders/2b/mpxyl1m54jvg2rcs2cnbsqxc0000gn/T/job3653979061400368429/build/resources/PrivacyInfo.xcprivacy
	at com.defold.extender.Extender.buildEngine(Extender.java:2244)
	at com.defold.extender.Extender.build(Extender.java:2612)
	at com.defold.extender.AsyncBuilder.asyncBuildEngine(AsyncBuilder.java:116)
	at jdk.internal.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:355)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:768)
	at org.springframework.aop.interceptor.AsyncExecutionInterceptor.lambda$invoke$0(AsyncExecutionInterceptor.java:113)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.nio.file.FileAlreadyExistsException: /var/folders/2b/mpxyl1m54jvg2rcs2cnbsqxc0000gn/T/job3653979061400368429/build/resources/PrivacyInfo.xcprivacy
	at java.base/sun.nio.fs.UnixCopyFile.copy(UnixCopyFile.java:573)
	at java.base/sun.nio.fs.UnixFileSystemProvider.copy(UnixFileSystemProvider.java:257)
	at java.base/java.nio.file.Files.copy(Files.java:1305)
	at com.defold.extender.Extender.buildPods(Extender.java:983)
	at com.defold.extender.Extender.buildEngine(Extender.java:2214)
	... 14 more
```

Repo case project:
[test_yandex.zip](https://github.com/user-attachments/files/17291197/test_yandex.zip)

Full build log:
[log.txt_yandex_sdk.zip](https://github.com/user-attachments/files/17291191/log.txt_yandex_sdk.zip)

@Lerg any idea?

--- a/server/src/main/java/com/defold/extender/Extender.java
+++ b/server/src/main/java/com/defold/extender/Extender.java
@@ -945,10 +945,10 @@ class Extender {
             return outputFiles;
         }
 
-        LOGGER.info("buildPods");
+        LOGGER.info("buildPods - compiling pod source files");
         Set<File> podSourceLookup = new HashSet<>();
         for (PodSpec pod : resolvedPods.pods) {
-            LOGGER.info("building {}", pod.name);
+            LOGGER.info("buildPods - compiling pod source files for {}", pod.name);
             // The source files of each pod will be compiled and built as a library.
             // We use the same mechanism as when building the extension and create a
             // manifest context for each pod
@@ -978,14 +978,24 @@ class Extender {
         resourcesBuildDir.mkdir();
         List<String> resources = resolvedPods.getAllPodResources(platform);
         for (String resource : resources) {
+            // example:
+            // source = CocoaPodsService/Pods/YandexMobileAds/PrivacyInfo.xcprivacy
+            // dest   = build/resources/YandexMobileAds/PrivacyInfo.xcprivacy
+            //
+            // resourceFile = CocoaPodsService/Pods/YandexMobileAds/PrivacyInfo.xcprivacy
+            // relativeFile = YandexMobileAds/PrivacyInfo.xcprivacy
+            // resourceDestFile = build/resources/YandexMobileAds/PrivacyInfo.xcprivacy
             File resourceFile = new File(resource);
+            File relativeFile = resolvedPods.podsDir.toPath().relativize(resourceFile.toPath()).toFile();
             if (resourceFile.isFile()) {
-                File resourceDestFile = new File(resourcesBuildDir, resourceFile.getName());
+                File resourceDestFile = new File(resourcesBuildDir, relativeFile.toString());
+                resourceDestFile.getParentFile().mkdirs();
                 Files.copy(resourceFile.toPath(), resourceDestFile.toPath());
                 outputFiles.add(resourceDestFile);
             }
             else {
-                File resourceDestDir = new File(resourcesBuildDir, resourceFile.getName());
+                File resourceDestDir = new File(resourcesBuildDir, relativeFile.toString());
+                resourceDestDir.mkdirs();
                 FileUtils.copyDirectory(resourceFile, resourceDestDir);
                 outputFiles.add(resourceDestDir);
             }
@@ -2447,6 +2457,10 @@ class Extender {
     private List<File> buildApple(String platform) throws ExtenderException {
         LOGGER.info("Building Apple specific code");
         List<File> outputFiles = new ArrayList<>();
+
+        File resourceDirectory = new File(buildDirectory, "resources");
+        List<File> resources = ExtenderUtil.listFiles(resourceDirectory, ".*");
+        outputFiles.addAll(resources);
 
         List<File> privacyManifests = new ArrayList<>();
         privacyManifests.addAll(ExtenderUtil.listFilesMatchingRecursive(uploadDirectory, "PrivacyInfo.xcprivacy"));

